### DISCUSSION TITLE
通过 `tsconfig.json` 中的 `paths` 配置路径别名, 在vue文件中无法导入scss文件

### DISCUSSION THREAD

**Discussion by tthzwq** [web-infra-dev/rsbuild#2327] [Q&A] [answered]
## 通过 `tsconfig.json` 中的 `paths` 配置路径别名
```vue
<!-- App.vue -->

<template>
</template>
<script setup lang="ts">
</script>
<style scoped lang="scss">
@import '~@/style/responsive-form.scss';
</style>
```
```ts
// tsconfig.json

{
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"]
    },
...
}
```
<img width="820" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/56644784/9eefe2af-5661-406f-8c32-770c0a0c33d8">

## 通过 `source.alias` 配置路径别名 能正常导入scss文件
```ts
// rsbuild.config.ts

export default defineConfig({
  source: {
    alias: {
      '@': './src',
    },
  },
```

**Comment by SoonIter** [comment, marked as answer]
应该没有这种用法，~是用来导入 node_modules 的三方包用的，你可以改为 `@import '@/style/a.scss';`

<img width="400" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/63f81453-2740-4aa9-9c51-2d921bb9303f">

https://webpack.js.org/loaders/css-loader/#url

如果你一定要这么用的话可以这么做

```diff
// rsbuild.config.ts
export default defineConfig({
  source: {
    alias: {
      '@': './src',
+     '~@': './src',
    },
  },
```
